sql/2021/pwa/top_manifest_icon_sizes.sql

```diff
@@ -0,0 +1,88 @@
+#standardSQL
+# Top manifest icon sizes
+CREATE TEMPORARY FUNCTION getIconSizes(manifest STRING)
+RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+  var $ = Object.values(JSON.parse(manifest))[0];
+  return $.icons.map(icon => icon.sizes);
+} catch (e) {
+  return null;
+}
+''';
+
+SELECT
+  "PWA Sites" AS type,
+  _TABLE_SUFFIX AS client,
+  size,
+  COUNT(0) AS freq,
+  total,
```

Comment:
Same optimization as the following query
```suggestion
  SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS total,
```

Reply:
Actually we can't do this in this query and took a bit of digging into why. We need to be very careful with this syntax - it can only be used when the following applies:
- You are not using `HAVING` to restrict to top 100 (or whatever). `QUALIFY` is safe to use, as applied after.
- You are not doing any joins (including `UNNEST`)
- You want the total to reflect the `WHERE` clause.

In this case we are using both `UNNEST` and `HAVING`. We could switch the `HAVING` to `QUALIFY` but still have the `UNNEST`. So can't use this short cut, and have to use a `JOIN` to a total table.

In fact this has identified that I was using this total shortcut incorrect in `top_manifest_display_values` so have fixed that.

I wonder how many other people are using this incorrectly?

Reply:
Yeah just learned that window functions are applied after HAVING:
```
FROM -> WHERE -> GROUP BY -> HAVING -> OVER -> QUALIFY
```

UNNEST generally isn't a problem in and of itself though, you just have to remember that you've fundamentally changed what you're querying now. A simple `COUNT(DISTINCT url)` can completely fix that problem though

Reply:
The problem is you are not doing a simple `COUNT(DISTINCT url)`, but a `SUM(COUNT(DISTINCT url))` so that's not necessarily distinct URLs.

Reply:
Resolving this as updated @rviscomi 's other concern on the COUNTs.